### PR TITLE
mavlink_log refactor take 3

### PR DESCRIPTION
**Pull request by julianoes**
Since we can't use the ioctl call on Snapdragon, let's move the mavlink_log calls from ioctls to the uORB interface.

This is a third PR and replaces #4018. The reason for yet another PR is that this one is on top of the branch feature_calib_refactor. This makes the development and testing easier for me.

### CODE REVIEW

**Comment by julianoes** [comment]
@LorenzMeier I added buffering back in but some messages are still dropped. I think this happens if a task with higher priority than mavlink publishes two messages right one after the other.

**Comment by julianoes** [comment]
@AndreasAntener Any chance you could give this an independent bench test? If it doesn't break anything, I'd say I'll go ahead and merge it.

**Comment by LorenzMeier** [comment]
@julianoes That corner case is expected and needs a uORB internal buffer.

**Comment by LorenzMeier** [comment]
Will bench test tomorrow morning.

**Comment by LorenzMeier** [comment]
I rebased and force-pushed. You need to delete your local branch and clone fresh if you want to try this.

**Comment by LorenzMeier** [comment]
The priority of the messages is sent incorrectly. Info messages are sent as alerts.

**Comment by julianoes** [comment]
Did you do the rebasing on feature_calib_refactor?

**Comment by julianoes** [comment]
Will fix the priorities, I was wondering about this before.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -360,58 +361,65 @@ class MavlinkStreamStatustext : public MavlinkStream
...
 
-						if (write_err_count >= write_err_threshold) {
-							(void)fclose(fp);
-							fp = nullptr;
-						} else {
-							(void)fputs("\n", fp);
-							(void)fsync(fileno(fp));
-						}
+// TODO: the logging doesn't work on Snapdragon yet because of file paths.
+#ifdef __PX4_POSIX_EAGLE
```
Comment (LorenzMeier) [inline src/modules/mavlink/mavlink_messages.cpp]: Looks like it should be #ifndef here, not #ifdef.

**Comment by julianoes** [comment]
@LorenzMeier: Let me please handle the rebasing of my branches, thanks.


> I rebased and force-pushed. You need to delete your local branch and clone fresh if you want to try this.

This sounds tedious, if a branch has been force-pushed I usually do it this way:
```
git fetch
git checkout feature_mavlinklog_refactor3
git reset --hard origin/feature_mavlinklog_refactor3
```

File: src/modules/mavlink/mavlink_messages.cpp (hunk 1 above)
Comment (julianoes) [inline src/modules/mavlink/mavlink_messages.cpp]: Nice catch, I'll fix it.

**Comment by LorenzMeier** [comment]
The printing should have been kept with printf / fprintf, because the formatting is now messed up on NSH:

```
nsh> WARN  telem> 
WARN  NOT ARMING: Press safety switch first!
WARN  telem> 
WARN  CAUTION: Avionics power low:   4.66 Volt
```

**Comment by julianoes** [comment]
Thanks @LorenzMeier both changes look good.

**Comment by LorenzMeier** [comment]
I've merged this into master, so we can close all these now.
